src/linker/implementation.py

```diff
@@ -42,6 +71,10 @@ def run(
             config=self.config,
         )
 
+        if self.requires_spark and computing_environment != "local":
+            logger.info(f"Shutting down spark cluster for pipeline step ID {step_id}")
+            session.control(job_id, drmaa.JobControlAction.TERMINATE)
```

Comment:
Nice! I didn't know stopping a job array was this easy.

Reply:
Indeed. Rajan pointed out that that's how we do it in psimulate.